src/arithmetic.rs

```diff
@@ -172,7 +173,7 @@ pub fn best_multiexp<C: CurveAffine>(coeffs: &[C::Scalar], bases: &[C]) -> C::Cu
         let chunk = coeffs.len() / num_cpus;
         let num_chunks = coeffs.chunks(chunk).len();
         let mut results = vec![C::Curve::identity(); num_chunks];
-        thread::scope(|scope| {
+        multicore::scope(|scope| {
             let chunk = coeffs.len() / num_cpus;
```

Comment:
This code chunks uniformly on `num_cpus` regardless of the size of the threadpool. The simple fix would be to make `NUM_CPUS` a `pub(crate)` static, and then use it here (and everywhere else in the crate we call `num_cpus::get()`).

Reply:
This has been addressed. I made it `pub` rather than `pub(crate)`, since there's no reason to hide it, and external code might need to use it.